### DISCUSSION TITLE
C++ port

### DISCUSSION THREAD

**Discussion by violanes** [SteamRE/SteamKit#1263] [General]
I've made a C++20 port of SteamKit recently which works on any platform and has only few necessary dependencies like boost and protobuf.
I think it's worth a mention in ports list as there are no c++ libraries written according to modern standards and most importantly maintained to this day.

[steam_cpp](https://github.com/violanes/steam_cpp)

**Comment by yaakov-h** [comment]
If you can post the exact text you want added to the [Ports](https://github.com/SteamRE/SteamKit/wiki/Ports) page, I can add it in.

**Comment by Christian-Stieber** [comment]
Also, while I don't consider it a SteamKit port and it's still heavily "work in progress", I have a C++ framework as well:
https://github.com/Christian-Stieber/Christians-Steam-Framework

There's also a CLI "bot" as an actual application for this:
https://github.com/Christian-Stieber/Christians-Steam-Bot

I always thought that C++ would be kind of dead for stuff like that, so I'm actually surprised to see someone else take a shot at it as well.

It officially supports Linux and Windows. The bot is slightly more involved due to the Windows-console vs. Linux-termios stuff, but the Framework itself has a couple minor platform-#ifdef as well, like in the MachineInfo section or when it needs to figure out where to store data files (homedir vs. appdata and associated naming conventions), so it's not going to run on MacOS "out of the box" (and I don't have a Mac...).

Again, keep in mind that these are work in progress. The main goal for myself is to replicate what I do with ASF, so on top of the general Steam communication stuff I've been focusing on stuff like trading bot inventories, "playing" games, getting badge data etc. Modules for clearing discovery queues and claiming sale stickers are also there, alongside some of the generic stuff like getting games/licenses on an account. I'm also somewhat working on notifications, but haven't figured out all the details yet...

It also supports some of the "mid-level" stuff, like making authenticated requests for webpages and client-side "unified messaging". It also takes care of the rate-limiting across all active clients -- arguably, the inventory rate limiting is a bit too much right now, but that was from before I learned actual numbers, and I've been too lazy to change it yet.

And, did I mention "work in progress"? Yeah, documentation is... lacking. Most modules have some comments that likely have different variations of usefulness, but there's no overarching manual.

All of this is in various stages of usefulness, and immediate plans are not even adding a card farming module, but to clean up the existing APIs a bit and improve how the bot CLI is done to allow the bot to better automate things that I'm doing manually.

Some of it is also mostly "designed" with a CLI application in mind, like the framework requesting passwords or 2FA codes. The API is considered a bit quick-and-dirty so I could work on more interesting things; I haven't tried making a GUI-based application yet to see what needs to be generalized. Also, while I thought about trying to make a console-QR printer, I'm not currently supporting QR codes on logins.
